data_and_models/pipelines/ner/clean.py

```diff
@@ -0,0 +1,131 @@
+# Blue Brain Search is a text mining toolbox focused on scientific use cases.
+#
+# Copyright (C) 2020  Blue Brain Project, EPFL.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+"""
+This script cleans annotations exported with Prodigy as .jsonl files.
+
+Run 'analyze.py' before to identify relevant cleaning rules.
+
+First, this script keeps only valid texts from the annotations (see next paragraph).
+Second, this script normalizes entity labels (use upper case for labels in lower case).
+Third, this script keeps only the given entity label and renames it if necessary.
+
+Valid texts are texts which have been accepted, have or not spans (i.e. entities), and
+have a unique 'input_hash'.
+
+The output is one file. The file is put in the same directory as the input file. It is
+named following the pattern 'annotations_<entity_label>.jsonl'.
+"""
+
+from argparse import ArgumentParser
+from collections import Counter
+from pathlib import Path
+from typing import Any, Dict, Set
+
+import srsly
+
+parser = ArgumentParser()
+parser.add_argument(
+    "input_path",
+    type=Path,
+    help="path of the annotations exported with Prodigy (.jsonl file)",
+)
+parser.add_argument(
+    "kept_label",
+    type=str,
+    help="keep only this entity label in the annotations (converted to upper case)",
+)
+parser.add_argument(
+    "--renamed_label",
```

Comment:
Actually, a small comment here.

I think  we generally use dashed in arguments instead of underscores, so in this case it would be `--renamed-label` (the parsed variable will still have the underscores anyway)

Reply:
Interesting.

I used `add_er.py` as an example. In add_er.py, the `underscores` are used:

https://github.com/BlueBrain/Search/blob/adb6e49d01db5e34f8816ebdc6b43d161f40e348/data_and_models/pipelines/ner/add_er.py#L41-L52

Reply:
There are actually a `lot` of other arguments using the `underscore` as a separator:

➡️ The following is only a `lower bound` as the regex looks only for the optional argument (starting with `--`). A regexp for looking for the positional arguments (not starting with --) is a bit more tricky.

```
$ grep -r --include "*.py" -E "\"--[a-z]+(_[a-z]+)+\"" .

./data_and_models/pipelines/ner/transformers_vs_spacy/spacy/eval.py:    "--annotation_files",
./data_and_models/pipelines/ner/transformers_vs_spacy/spacy/eval.py:    "--output_file",

./data_and_models/pipelines/ner/eval.py:    "--annotation_files",
./data_and_models/pipelines/ner/eval.py:    "--output_file",

./data_and_models/pipelines/ner/add_er.py:    "--output_file",
./data_and_models/pipelines/ner/add_er.py:    "--patterns_file",

./data_and_models/pipelines/ner/interrater.py:    "--output_dir",

./data_and_models/pipelines/sentence_embedding/train.py:    "--sentences_file",
./data_and_models/pipelines/sentence_embedding/train.py:    "--output_dir",
./data_and_models/pipelines/sentence_embedding/eval.py:    "--annotation_files",
./data_and_models/pipelines/sentence_embedding/eval.py:    "--output_dir",

./benchmarks/conftest.py:    parser.addoption("--embedding_server", default="", help="Embedding server URI")
./benchmarks/conftest.py:    parser.addoption("--mining_server", default="", help="Mining server URI")
./benchmarks/conftest.py:    parser.addoption("--mysql_server", default="", help="MySQL server URI")
./benchmarks/conftest.py:    parser.addoption("--search_server", default="", help="Search server URI")
./benchmarks/conftest.py:        "embedding_server": request.config.getoption("--embedding_server"),
./benchmarks/conftest.py:        "mining_server": request.config.getoption("--mining_server"),
./benchmarks/conftest.py:        "mysql_server": request.config.getoption("--mysql_server"),
./benchmarks/conftest.py:        "search_server": request.config.getoption("--search_server"),
```

Reply:
Yes, unfortunately this was not caught in the previous PRs. But maybe at least we can do it right in the code. If we have the time resourses obviously it would be cool to correct all other ones as well.

Reply:
After some investigations:

1 - 



# Positional argument https://bugs.python.org/issue15125

Reply:
After some investigations:

1 - The convention for the separator in `positional` arguments is to use an `underscore`.

2 - The convention for the separator in `optional` arguments is to use an `dash`.

3 - `argparse` do `not` automatically convert a `dash` separator inside `positional` arguments ([bug report](https://bugs.python.org/issue15125)).

4 - git ([example](https://git-scm.com/docs/git-checkout)), DVC ([example](https://dvc.org/doc/command-reference/diff)), spaCy ([example](https://spacy.io/api/cli#train)) follows the conventions above.

➡️ Commit f94cfdd applies the conventions above.

Reply:
Yes, exactly. It's true they could've implemented this for positional arguments too. But probably the motivation is that the user never has to type out the name of the positional argument, while for optional arguments they do. Anyway, would be nice for consistency nonetheless.

Anyway, thanks for changing this.